src/main/home/send/SendAmount/index.tsx

```diff
@@ -30,40 +34,82 @@ type SendAmountScreenProps = NativeStackScreenProps<
 >;
 
 const SendAmountScreen = ({ route }: SendAmountScreenProps) => {
-  const { amount, address, username } = route.params;
+  const { amount, address, username, wallet, sender } = route.params;
   const { t } = useTranslation();
   const navigation = useNavigation();
   const isDarkMode = useColorScheme() === 'dark';
-  const wallet = useWallet();
   const [quaiBalance, setQuaiBalance] = React.useState(0);
   const [hideBalance, setHideBalance] = React.useState(false);
   const inputRef = useRef<TextInput>(null);
   const { eqInput, input, keyboard, onSwap } = useAmountInput(`${amount}`);
 
   const backgroundStyle = {
     backgroundColor: isDarkMode ? styledColors.black : styledColors.light,
-    width: '100%',
-    height: '100%',
-  };
-
-  const textColor = {
-    color: isDarkMode ? styledColors.white : styledColors.black,
+    flex: 1,
   };
 
   const equivalentUnitTextColorStyle = {
     color: isDarkMode ? styledColors.gray : styledColors.black,
   };
 
   const goToTip = () => {
+    const amountInUSD =
+      input.unit === Currency.USD ? input.value : eqInput.value;
+    const amountInQUAI =
+      input.unit === Currency.QUAI ? input.value : eqInput.value;
+    if (!input.value) {
+      // TODO: show friendly error message
+      return;
+    }
+    if (Number(amountInQUAI) > quaiBalance) {
+      // TODO: show friendly error message
+      // return;
+    }
+
     // @ts-ignore
     navigation.navigate('SendStack', {
       screen: 'SendTip',
       params: {
```

Comment:
with the current proper typing you could now:

1. Remove `useNavigation` and acquire `navigation` from props
2. Remove the ts-ignore line
3. Refactor this to make it work

```suggestion
    navigation.navigate('SendTip', {
      params: {
```